File: app/src/main/java/com/danielmaia/desafioandroid/database/RepoDatabase.java
package com.danielmaia.desafioandroid.database;

import android.content.Context;

import androidx.room.Database;
import androidx.room.Room;
import androidx.room.RoomDatabase;

import com.danielmaia.desafioandroid.dao.PullDao;
import com.danielmaia.desafioandroid.dao.RepoDao;
import com.danielmaia.desafioandroid.model.Pull;
import com.danielmaia.desafioandroid.model.Repo;

@Database(entities = {Repo.class, Pull.class}, version = 1, exportSchema = false)
public abstract class RepoDatabase extends RoomDatabase {

    private static final String DB_NAME = "repo.db";
    private static volatile RepoDatabase instance;

    public static synchronized RepoDatabase getInstance(Context context) {
        if (instance == null) {
            instance = create(context);
        }
        return instance;
    }

    private static RepoDatabase create(final Context context) {
        return Room.databaseBuilder(
                context,
                RepoDatabase.class,
                DB_NAME).build();
    }

    public abstract RepoDao getRepoDao();
    public abstract PullDao getPullDao();

}

File: app/src/main/java/com/example/desafioandroid/data/database/RepoDatabase.kt
package com.example.desafioandroid.data.database

import androidx.room.Database
import androidx.room.RoomDatabase
import com.example.desafioandroid.data.database.dao.RepoDao
import com.example.desafioandroid.data.database.entities.RepoEntity

@Database(entities = [RepoEntity::class], version = 1)
abstract class RepoDatabase : RoomDatabase() {

    abstract fun getRepoDao(): RepoDao
}